docs: add getting started and clean up docs (#2490)
lyne-design-system/lyne-components: docs/GETTING_STARTED.md

# Getting Started

Basic steps to integrate the lyne components into your own project.

Choose your technology to receive the first steps to take.

<details>
  <summary>Plain Javascript</summary>

1. Install the `@sbb-esta/lyne-components` package:

   ```sh
   npm install --save @sbb-esta/lyne-components
   ```

   or, if using yarn:

   ```sh
   yarn add @sbb-esta/lyne-components
   ```

2. Including typography is required to apply all SBB styles to your application.

   ```css
   @import 'node_modules/@sbb-esta/lyne-components/typography.css';
   ```

3. Import the desired element and add it to globalThis:

   ```ts

>    import { SbbButtonElement } from '@sbb-esta/lyne-components/button';

Author: check other occurrences as well

```suggestion
   import { SbbButtonElement } from '@sbb-esta/lyne-components/button.js';
```

> IMPORTANT: CSS variable names which are not documented in the component docs section, can be changed at any time.

Please reach out to the Lyne team if you think a component needs a new variant.

### Animations

Generally, animations are part of the design and should not be suppressed.
However, e.g. for testing purposes, it makes sense to deactivate them.

Add the `sbb-disable-animation` CSS class to disable animations and transition effects for the element and all its children.

Sometimes, you might need to disable animations only for a specific element.
To achieve that, you can add the `sbb-disable-animation-locally` class or re-enable animations using the `sbb-enable-animation` CSS class.

```html
<sbb-component class="sbb-disable-animation-locally">
  <!-- animations will play -->
</sbb-component>

<!-- Or  alternatively-->

<sbb-component class="sbb-disable-animation">
  <sbb-component class="sbb-enable-animation">
    <!-- animations will play -->
  </sbb-component>
</sbb-component>
```

### Units

All our tokens and components use `rem` as size unit.

> We strongly recommend using the `rem` unit all over the project to guarantee consistent scaling

Reviewer: ```suggestion
We strongly recommend using the `rem` unit to guarantee consistent scaling
```

when font size changes in browsers.

### Stacking

> As we can't use popover API so far, stacking of overlay context is done manually.

Reviewer: ```suggestion
As we can't use popover API yet, stacking of overlay context is done manually.
```

However, this can interfere with the z-index of your components.
Therefore, every overlay component provides a CSS variable to override its z-index.
Additionally, there is the global CSS variable `--sbb-overlay-default-z-index` that has a default z-index of 1000.

> So developers have the chance to either globally or component-specifically change the z-index.

Reviewer: ```suggestion
With this developers have the chance to change the z-index either globally or on component level.
```
